OMCompiler/Compiler/NFFrontEnd/NFSimplifyExp.mo

```diff
@@ -783,8 +784,20 @@ algorithm
       algorithm
         tb := simplify(tb);
         fb := simplify(fb);
+
+        if Expression.isEqual(tb, fb) then
+          // if cond then x else x => x
+          ifExp := tb;
+        elseif Expression.isBoolean(tb) and Expression.isBoolean(fb) then
+          // if cond then true else false => cond
+          // if cond then false else true => not cond
```

Comment:
Looking at the testsuite, I believe you can generalize that to
```suggestion
          // if cond1 then true else cond2 => cond1 or cond2
          // if cond1 then false else cond2 => (not cond1) and cond2
          // if cond1 then cond2 else true => (not cond1) or cond2
          // if cond1 then cond2 else false => cond1 and cond2
```

Reply:
Sure, I just implemented the same rule that the old frontend uses since it was trivial, we could simplify such expressions even further if we wanted to.